src/py42/clients/util.py

```diff
@@ -2,12 +2,13 @@
 
 
 def get_all_pages(func, key, *args, **kwargs):
-
-    item_count = settings.items_per_page
     page_num = 0
-    while item_count >= settings.items_per_page:
+    if kwargs.get("page_size") is None:
+        kwargs["page_size"] = settings.items_per_page
+    item_count = kwargs["page_size"]
+    while item_count >= kwargs["page_size"]:
```

Comment:
Something is fishy here, can't understand the intention!!
 `item_count` is set to `page_size` on line 8 then on line 9 while condition would be satisfied on equality condition.  
I think the while loop will always get executed only once, then do we need the while loop?

Reply:
a while loop continues for each iteration that is satisfied, what you're saying would be true if `item_count` was _not_ equal to `kwargs["page_size"].`

`item_count` gets its value changed at the end of this loop to be equal to the number of items that the page returned. In other words, this will loop until we get to page that returns fewer items than `kwargs["page_size"]`, which is what we want.

Reply:
Thank you.